scripts/filter_techniques_by_category.py

```diff
@@ -0,0 +1,180 @@
+    parser.add_argument("-o", "--operation", type=str, required=True, choices=["groups", "mitigations"], help="Operation to perform on ATT&CK content.")
+    parser.add_argument("-s", "--save", type=str, required=False, help="Save the CSV file with a different filename.")
+    return parser
+
+
+def do_groups(ds):
+    """Main logic to match techniques to groups"""
+    all_attack_patterns = get_all_techniques(ds)
+    writable_results = []
+
+    for attack_pattern in all_attack_patterns:
+        tid = grab_external_id(attack_pattern)
+
+        # Grabs uses relationships for identified techniques
+        relationships = filter_for_term_relationships(ds, "uses", attack_pattern.id, source=False)
+
+        for relationship in relationships:
+            # Groups are defined in STIX as intrusion-set objects
+            groups = filter_by_type_and_id(ds, "intrusion-set", relationship.source_ref)
+
+            if groups:
+                group = groups[0]
+                gid = grab_external_id(group)
+                writable_results.append(
+                    {
+                        "TID": tid,
+                        "Technique Name": attack_pattern.name,
+                        "GID": gid,
+                        "Group Name": group.name,
+                        "Group Description": group.description,
+                        "Usage": relationship.description
+                    }
+                )
+    return sorted(writable_results, key=lambda x: (x["TID"], x["GID"]))
+
+
+def do_mitigations(ds):
+    """Main logic to match techniques to mitigations"""
+    all_attack_patterns = get_all_techniques(ds)
+    writable_results = []
+
+    for attack_pattern in all_attack_patterns:
+        tid = grab_external_id(attack_pattern)
+
+        # Grabs mitigation relationships for identified techniques
+        relationships = filter_for_term_relationships(ds, "mitigates", attack_pattern.id, source=False)
+
+        for relationship in relationships:
+            # Mitigations are defined in STIX as course-of-action objects
+            mitigation = filter_by_type_and_id(ds, "course-of-action", relationship.source_ref)
+
+            if mitigation:
+                mitigation = mitigation[0]
+                mid = grab_external_id(mitigation)
+                writable_results.append(
+                    {
+                        "TID": tid,
+                        "Technique Name": attack_pattern.name,
+                        "MID": mid,
+                        "Mitigation Name": mitigation.name,
+                        "Mitigation Description": escape_chars(mitigation.description),
+                        "Application": escape_chars(relationship.description),
+                    }
+                )
+    return sorted(writable_results, key=lambda x: (x["TID"], x["MID"]))
```

Comment:
`do_mitigations` and `do_groups` are nearly identical in functionality. Might be worth refactoring into a single function to adhere to good [DRY principles](https://www.wikiwand.com/en/Don%27t_repeat_yourself).